include/swift/AST/AnyRequest.h

```diff
@@ -0,0 +1,227 @@
+//===--- AnyRequest.h - Requests Instances ----------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file defines the AnyRequest class, which describes a stored request.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_AST_ANYREQUEST_H
+#define SWIFT_AST_ANYREQUEST_H
+
+#include "swift/Basic/TypeID.h"
+#include "llvm/ADT/DenseMapInfo.h"
+#include "llvm/ADT/Hashing.h"
+
+namespace swift {
+
+using llvm::hash_code;
+using llvm::hash_value;
+
+class DiagnosticEngine;
+
+/// Stores a request (for the \c Evaluator class) of any kind.
+///
+/// Requests must be value types and provide the following API to be stored in
+/// an \c AnyRequest instance:
+///
+///   - Copy constructor
+///   - Equality operator (==)
+///   - Hashing support (hash_value)
+///   - TypeID support (see swift/Basic/TypeID.h)
+///   - Cycle diagnostics operations:
+///
```

Comment:
Did you forget 'OutputType breakCycle() const' here? It looks like it's present in a similar comment down below.

Reply:
`AnyRequest` itself doesn't need it, because we only call the `breakCycle` operation on a request before it's been type-erased. We can add it here if we need it later.